packages/shared/tests/hasProtocal.spec.ts

```diff
@@ -0,0 +1,24 @@
+import { describe, expect, it } from 'vitest'
+import { hasProtocol } from '../src/index.js'
+
+const testCases: [string, ReturnType<typeof hasProtocol>][] = [
+  ['ftp://foobar.com', true],
+  ['mailto:foobar', true],
+  ['tel:foobar', true],
+  ['https://foobar.com', true],
+  ['http://foobar.com', true],
+  ['foobar.com', false],
+  ['/foo/bar', false],
+  ['../foo/bar', false],
+  ['//foobar.com', false],
+]
```

Comment:
Should also include some custom protocols in the test cases

Reply:
ftp is already a test case, we no longer treat it specially

Reply:
I mean those special protocols, like `ms-windows-store` mentioned in #1404